### PR TITLE
Check link ordinal to make sure it is targetted  for foreign function

### LINKED ISSUE #100009: link_ordinal attribute position is not restricted
The `link_ordinal` attribute currently does not appear to have any restrictions on where it is located.

The following code compiles without warning or error:

```rust
#[link_ordinal(123)]
struct Foo{}
```

I expected to see this happen: An error when `link_ordinal` is placed in the wrong position.

Instead, this happened: `link_ordinal` is ignored in wrong positions.

The code in [`CheckAttrVisitor`](https://github.com/ehuss/rust/blob/4be9a1b6ac67d03776b1e2a420c95a3b8348356d/compiler/rustc_passes/src/check_attr.rs#L63) can be used to enforce this.  Just add a `check_link_ordinal` method that validates its location (see the other methods for how to do that).

### Meta
```
rustc 1.64.0-nightly (2643b1646 2022-07-27)
binary: rustc
commit-hash: 2643b16468fda787470340890212591d8bc832b7
commit-date: 2022-07-27
host: x86_64-apple-darwin
release: 1.64.0-nightly
LLVM version: 14.0.6
```

### PR DESCRIPTION
**Pull request by chenyukang**
Fix #100009, when link ordinal is not target for foreign functions, emit an error.

cc @dpaoliello

### CODE REVIEW

File: compiler/rustc_passes/src/check_attr.rs [hunk 1]
```diff
@@ -1861,6 +1862,16 @@ impl CheckAttrVisitor<'_> {
         }
     }
 
+    fn check_link_ordinal(&self, attr: &Attribute, _span: Span, target: Target) -> bool {
+        match target {
+            Target::ForeignFn => true,
```
Comment (dpaoliello) [inline compiler/rustc_passes/src/check_attr.rs:1867]: `link_ordinal` can also be applied to `ForeignStatic`: #100073

Comment (chenyukang) [reply compiler/rustc_passes/src/check_attr.rs]: Fixed and add testcase for it.

**Review by dpaoliello — APPROVED** [review]

**Comment by michaelwoerister** [comment]
Looks great, thanks for the PR, @chenyukang!

@bors r+

**Pull request merged**